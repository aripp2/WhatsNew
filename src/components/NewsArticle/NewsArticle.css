img {
  height: 250px;
  width: 100%;
  margin: 0;
}

article {
  width: 45%;
  margin: 20px;
  background-color: white;
}

h3 {
  height: 100px;
  font-size: 1.5em;
  color: #141527;
  margin-top: 0;
  padding: 5px;
}

p {
  height: 150px;
  font-size: 1.2em;
  color: #141527;
  padding: 10px;
}

footer {
  height: 50px;
  width: 100%;
  color: white;
  font-size: 1.5em;
  background-color: grey;
  display: flex;
  justify-content: center;
  align-items: center;
}