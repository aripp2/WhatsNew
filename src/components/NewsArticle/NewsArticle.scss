@import './src/variables.scss';

img {
  height: 250px;
  width: 100%;
  margin: 0;
  border-radius: 3px 3px 0 0;
}

article {
  width: 45%;
  margin: 20px;
  background-color: white;
  border-radius: 3px;
  border: 3px outset $mid-color;
}

h3 {
  height: 100px;
  font-size: 1.5em;
  color: $dark-color;
  margin-top: 0;
  padding: 5px;
}

p {
  height: 175px;
  font-size: 1.2em;
  color: $dark-color;
  padding: 10px;
}

a {
  text-decoration: none;
    display: flex;
  justify-content: center;
  align-items: center;
  &:hover {
    color: $dark-color;
    svg {
      fill: $dark-color;
    }
  }
}

footer {
  height: 50px;
  width: 100%;
  color: $highlight-color;
  font-size: 1.5em;
  background-color: $mid-color;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 0 0 3px 3px;
}

svg {
  height: 1em;
  width: 1em;
  fill: $highlight-color;
  margin-left: 25px;
}