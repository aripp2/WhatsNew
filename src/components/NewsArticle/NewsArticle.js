import React from 'react';
import './NewsArticle.scss';
import { FaArrowRight } from 'react-icons/fa';

const NewsArticle = (props) => {
  return (
    <article>
      <img src={props.image} alt=""/>
      <h3>{props.headline}</h3>
      <p>{props.description}</p>
      <footer>
        <a href={props.url}>

        See full article
        <FaArrowRight />
        </a>
      </footer>
    </article>
  )
}

export default NewsArticle;