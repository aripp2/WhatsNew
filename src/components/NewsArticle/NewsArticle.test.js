import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<NewsArticle
      image="url"
      headline="This is the headline"
      description="This is the description"
      url="link"
    />);

    expect(wrapper).toMatchSnapshot();
  });
});   

