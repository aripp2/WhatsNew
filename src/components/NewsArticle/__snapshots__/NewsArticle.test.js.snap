// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed in correctly 1`] = `
<article>
  <img
    alt=""
    src="url"
  />
  <h3>
    This is the headline
  </h3>
  <p>
    This is the description
  </p>
  <footer>
    <a
      href="link"
    >
      See full article
    </a>
  </footer>
</article>
`;
